python/ray/_private/ray_logging/logging_config.py

```diff
@@ -88,6 +91,31 @@ def _get_dict_config(self) -> dict:
         """
         return _dict_config_provider.get_dict_config(self.encoding, self.log_level)
 
+    def _setup_log_record_factory(self):
+        old_factory = logging.getLogRecordFactory()
+
+        def record_factory(*args, **kwargs):
+            record = old_factory(*args, **kwargs)
+            # Python logging module starts to use `time.time_ns()` to generate `created`
+            # from Python 3.13 to avoid the precision loss caused by the float type.
+            # Here, we generate the `created` for the LogRecord to support older Python
+            # versions.
+            ct = time.time_ns()
+            record.created = ct / 1e9
+
+            from ray._private.ray_logging.constants import LogKey
+
+            record.__dict__[LogKey.TIMESTAMP_NS.value] = ct
+
+            return record
+
+        logging.setLogRecordFactory(record_factory)
+
+    def _apply(self):
+        """Set up both the LogRecord factory and the logging configuration."""
+        self._setup_log_record_factory()
```

Comment:
`_apply` is the only place where `_setup_log_record_factory` and `_get_dict_config` are called. Should we remove `_setup_log_record_factory` and `_get_dict_config` and move their code directly into `_apply`? The class will look more concise, but it may not be as easy to test each part separately. cc @jjyao

Reply:
Works for me